clash-prelude/src/Clash/Explicit/Moore.hs

```diff
@@ -44,14 +45,14 @@ import           Clash.XException                 (Undefined)
 -- macT s (x,y) = x * y + s
 --
 -- mac
---   :: 'Clock' mac Source
---   -> 'Reset' mac Asynchronous
---   -> 'Signal' mac (Int, Int)
---   -> 'Signal' mac Int
--- mac clk rst = 'moore' clk rst macT id 0
+--   :: 'Clock' tag
```

Comment:
missing `KnownDomain tag dom` constraint, probably missing in other places as well.